utils/error_collection.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ErrorCollection knows how to combine multiple error strings
+type ErrorCollection []error
+
+// Add puts given error to collection
+func (ec *ErrorCollection) Add(errors ...error) {
+	*ec = append(*ec, errors...)
+}
+
+// String concatenates collection to single string
+func (ec *ErrorCollection) String() string {
+	return ec.Stringf("ErrorCollection: %s", ", ")
+}
+
+// Stringf concatenates collection to single string with item formation
+func (ec *ErrorCollection) Stringf(format, errorDelimiter string) string {
+	errorStrings := make([]string, 0)
+	for _, err := range *ec {
+		errorStrings = append(errorStrings, err.Error())
```

Comment:
The current implementation allows adding `nil` error, which will panic on `err.Error()`. 
We need to protect it somehow, in `Add` method or here.

Reply:
Protected in `Add()` method